transport-native-epoll/src/main/java/io/netty/channel/epoll/AIOContext.java

```diff
@@ -109,13 +138,40 @@ public void processReady(AIOEpollFileChannel file) {
                     throw new IllegalStateException("" + id + " missing");
                 }
 
+                // Finally complete the request
                 int len = (int) lengthRead;
                 assert len == lengthRead;
                 assert len >= 0 : len;
                 request.buffer.limit(len);
                 request.buffer.position(len);
                 request.handler.completed(len, request.attachment);
             }
+
+            //add any pending reads
+            //We do this after processing the finished reads incase any outstanding reads
+            //they share the same read buffer
+            for (int i = 0; i < numReady; i++) {
+                --outstanding;
+                IORequest req = null;
+                // Push any pending requests if we have room
+                try {
+                    if (outstanding + 1 <= maxOutstanding && (req = pendingRequests.poll()) != null) {
```

Comment:
Any other exception not "Resource temporarily unavailable" and `req` is silently dropped. I think this method should be split in two methods, one to process the requests that have completed and one to send more pending requests, and each method should have its own exception handling.